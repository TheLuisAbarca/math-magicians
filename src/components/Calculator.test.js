import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import CalculatorRender, { Calculator } from './Calculator';

describe('App Component: Calculator', () => {
  test('Verify Component is correctly rendered.', () => {
    const calculatorComponent = renderer.create(<CalculatorRender />);
    expect(calculatorComponent).toMatchSnapshot();
  });

  test('Verify Calculator is correctly displaying results with operation (3.3 x 3 = 9.9)', () => {
    const { getByText } = render(<Calculator />);
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('.'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('x'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    getByText('9.9');
  });
});
