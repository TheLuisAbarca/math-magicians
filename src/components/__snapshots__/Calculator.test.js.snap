// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component: Calculator Verify Component is correctly rendered. 1`] = `
<section
  className="calculator-section"
>
  <h2
    className="section-title"
  >
    Let's do some math
  </h2>
  <div
    className="calculator-content"
  >
    <div
      className="calculator"
    >
      <div
        className="calculator-display"
      >
        <div
          className="display-value"
        >
          0
        </div>
      </div>
      <div
        className="calculator-keypad"
      >
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          className="keypad-button span-2-col || ''"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`App Component: Calculator renders page correctly 1`] = `
<section
  className="calculator-section"
>
  <h2
    className="section-title"
  >
    Let's do some math
  </h2>
  <div
    className="calculator-content"
  >
    <div
      className="calculator"
    >
      <div
        className="calculator-display"
      >
        <div
          className="display-value"
        >
          0
        </div>
      </div>
      <div
        className="calculator-keypad"
      >
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          className="keypad-button span-2-col || ''"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="keypad-button undefined || ''"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="keypad-button bg-orange || ''"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</section>
`;
